monero-sys/src/lib.rs

```diff
@@ -1570,7 +1616,121 @@ impl FfiWallet {
         self.address(Self::MAIN_ACCOUNT_INDEX, 0)
     }
 
-    /// Initialize the wallet and download initial values from the remote node.
+    /// Get the address for the given account and subaddress index.
+    pub fn address_at(&self, account_index: u32, address_index: u32) -> monero::Address {
+        // Reuse the private `address` helper
+        self.address(account_index, address_index)
+    }
+
+    /// Get the number of subaddresses for a given account.
+    pub fn num_subaddresses(&self, account_index: u32) -> usize {
+        ffi::numSubaddresses(&self.inner, account_index) as usize
+    }
+
+    /// Get the label for a specific subaddress.
+    pub fn subaddress_label(&self, account_index: u32, address_index: u32) -> String {
+        ffi::getSubaddressLabel(&self.inner, account_index, address_index)
+            .context("Failed to get subaddress label: FFI call failed with exception")
+            .expect("getSubaddressLabel should not fail")
```

Comment:
We've got to be very careful with these because one panic will bring the whole app down